### ISSUE TITLE
Dependencies on common certificates

### ISSUE THREAD

**Issue by simewu** [litespeedtech/lsquic#300]
Hello,

I've been interested in the QUIC protocol [documentation](https://datatracker.ietf.org/doc/html/rfc9000), and have specifically been focusing on the security of the digital certificates within each handshake.

I noticed your source code contains four predefined certificates, known as "common certificates":
- Common cert 2a: https://github.com/litespeedtech/lsquic/blob/master/src/liblsquic/common_cert_set_2a.inc
- Common cert 2b: https://github.com/litespeedtech/lsquic/blob/master/src/liblsquic/common_cert_set_2b.inc
- Common cert 3a: https://github.com/litespeedtech/lsquic/blob/master/src/liblsquic/common_cert_set_3a.inc
- Common cert 3b: https://github.com/litespeedtech/lsquic/blob/master/src/liblsquic/common_cert_set_3b.inc

I was curious about what the functions are of these, and (if these are used), how can I supplement them for external certificates? I understand that _http_client_ has the _-C_ parameter to specify other certificates to use, but I have not had much luck with this.

Using _gdb_ for debugging, I've been able to show that these common certificate files are being loaded, but I don't know if they are being actively used, so I wanted to ask you guys directly.

Any help would be greatly appreciated.

**Comment by dtikhonov** [comment]
Yes, these common certificate files are still being used in the codebase for gQUIC (Google QUIC) connections.

**What they are:**
The common certificate sets (2a, 2b, 3a, 3b) are used for certificate compression in gQUIC protocol versions (Q046, Q050). This is an optimization that allows the client and server to reference commonly-used CA certificates by hash instead of transmitting the full certificate chain, reducing handshake overhead.

**Current usage:**
- The files are compiled into `lsquic_crt_compress.c`
- They are initialized via `lsquic_crt_init()` when the handshake module starts up
- The certificate compression code is actively used for gQUIC handshakes

**Key distinction:**
These common certificates are **only used for gQUIC** (the original Google QUIC protocol). They are **not used for IETF QUIC** (RFC 9000), which uses standard TLS 1.3 certificate handling instead.

Since lsquic supports both gQUIC and IETF QUIC, the common cert functionality remains active for backward compatibility with gQUIC connections.

**Regarding the -C parameter:**
The `-C` parameter in http_client is for specifying a trusted CA certificate file for validating the server's certificate - this is separate from the common certificate compression mechanism. The common certs are built-in and don't need to be (and can't be) supplemented via command-line parameters.
